File: examples/python/core/parallel/simple-parallel.py
from aws_durable_execution_sdk_python import (
    BatchResult,
    DurableContext,
    durable_execution,
)

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Process multiple services in parallel."""
    
    def check_inventory(ctx: DurableContext) -> dict:
        return ctx.step(lambda _: {"service": "inventory", "status": "ok"})
    
    def check_payment(ctx: DurableContext) -> dict:
        return ctx.step(lambda _: {"service": "payment", "status": "ok"})
    
    def check_shipping(ctx: DurableContext) -> dict:
        return ctx.step(lambda _: {"service": "shipping", "status": "ok"})
    
    # Execute all checks in parallel
    result: BatchResult[dict] = context.parallel([
        check_inventory,
        check_payment,
        check_shipping,
    ])
    
    return {
        "total": result.total_count,
        "successful": result.success_count,
        "results": result.get_results(),
    }

File: examples/java/core/parallel/simple-parallel.java
// Execute all checks concurrently using async child contexts
var inventoryFuture = ctx.runInChildContextAsync("check-inventory", Map.class,
    child -> child.step("check", Map.class,
        stepCtx -> Map.of("service", "inventory", "status", "ok")));

var paymentFuture = ctx.runInChildContextAsync("check-payment", Map.class,
    child -> child.step("check", Map.class,
        stepCtx -> Map.of("service", "payment", "status", "ok")));

var shippingFuture = ctx.runInChildContextAsync("check-shipping", Map.class,
    child -> child.step("check", Map.class,
        stepCtx -> Map.of("service", "shipping", "status", "ok")));

// Wait for all to complete
DurableFuture.allOf(inventoryFuture, paymentFuture, shippingFuture);

var inventory = inventoryFuture.get();
var payment = paymentFuture.get();
var shipping = shippingFuture.get();

return Map.of(
    "inventory", inventory,
    "payment", payment,
    "shipping", shipping);
